### PR TITLE
feat(progress-stepper)!: 1:1

### LINKED ISSUE #2052: [1:1] pfe-progress-stepper
# Component name: `<pfe-progress-stepper>`

Formerly known as `<pfe-progress-steps>`

## Design specification updates
[PatternFly v4 Progress Stepper](https://www.patternfly.org/v4/components/progress-stepper)

depends on [Popover](https://github.com/patternfly/patternfly-elements/issues/2291)

### PR DESCRIPTION
**Pull request by bennypowers**
## What I did
- remove `pfe-progress-steps` and add `pfe-progress-stepper`

closes #2052 
closes #2128
depends on #2296

### CODE REVIEW

**Review by marionnegp — COMMENTED** [review]
Took a quick look today and found a few things.

- [ ] The circles should be 24px (not including border) based on PF's. I'm guessing they're larger in PFE because of it's using rems.
- [ ] Circles in the compact version should be 14px (not including border).
- [ ] Icons in the circles aren't showing up for me.

**Comment by bennypowers** [comment]
> Took a quick look today and found a few things.
> 
>     * [ ]  The circles should be 24px (not including border) based on PF's. I'm guessing they're larger in PFE because of it's using rems.
> 
>     * [ ]  Circles in the compact version should be 14px (not including border).
They use the same rem value as pfv4, and the same token name, so I think these are unrelated
```css
--pf-c-progress-stepper--m-compact__step-icon--Width: 1.125rem;
```

File: elements/pfe-progress-stepper/pfe-progress-step.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      <div id="main" class="${classMap({ compact })}">
+        <slot id="title"></slot>
+        <slot id="description" name="description" ?hidden="${!hasDescription}">${this.description}</slot>
+      </div>
+    `;
+  }
+
+  updated(changed: PropertyValues<this>) {
+    super.updated?.(changed);
+    this.#internals.ariaCurrent = String(!!this.current);
```
Comment (brianferry) [inline elements/pfe-progress-stepper/pfe-progress-step.ts:78]: Looks like we need to wrap this with a checker to make sure we don't get an infinite loop here, it's currently causing the demos not to load for me.

File: elements/pfe-progress-stepper/pfe-progress-step.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  }
+}
```
Comment (zeroedin) [inline elements/pfe-progress-stepper/pfe-progress-step.ts:80]: As brian said, there is an issue here causing an infinite loop.

Here was my fix, although I'm not convinced it is ideal either
```Suggestion
  attributeChangedCallback(name:string, _old:string | null, value:string | null) {
    super.attributeChangedCallback(name, _old, value);
    if (name === 'current') {
      this.#internals.ariaCurrent = String(!!this.current);
    }
  }
```

File: elements/pfe-progress-stepper/pfe-progress-step.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  danger: { icon: 'circle-exclamation' },
+  warning: { icon: 'triangle-exclamation' },
+  info: { icon: 'resources-full', set: 'patternfly' },
+}));
+
+/**
+ * @slot -
+ *       Short description of the current step.
+ * @slot description
+ *       Longer decscription of the current step.
```
Comment (zeroedin) [inline elements/pfe-progress-stepper/pfe-progress-step.ts:24]:
```Suggestion
 *       Longer description of the current step.
```

File: elements/pfe-progress-stepper/pfe-progress-stepper.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+  /** Whether to use the compact layout */
+  @observed(function(this: PfeProgressStepper) {
+    this.querySelectorAll('pfe-progress-step').forEach(step => step.requestUpdate());
+  })
+  @property({ type: Boolean, reflect: true }) compact = false;
+
+  #internals = new InternalsController(this, {
+    role: 'progressbar',
+    ariaValueNow: this.value,
```
Comment (zeroedin) [inline elements/pfe-progress-stepper/pfe-progress-stepper.ts:37]: ariaValueNow needs a `string` value here.

```Suggestion
    ariaValueNow: this.value.toString(),
```

**Review by zeroedin — COMMENTED** [review]
Other than the one show stopper, and a couple of other small fixes, this is looking great!

File: elements/pfe-progress-stepper/pfe-progress-step.ts (hunk 1 above)
Comment (bennypowers) [reply elements/pfe-progress-stepper/pfe-progress-step.ts]: I wasn't able to reproduce this infinite loop, neither on the dev server, nor on the 11ty demo page.

Nonetheless, I pushed a commit which gates the assignment. PTAL

File: elements/pfe-progress-stepper/pfe-progress-step.ts (hunk 2 above)
Comment (bennypowers) [reply elements/pfe-progress-stepper/pfe-progress-step.ts]: probably better just to check for changes in `updated`, lit likes to take control of ce lifecycle callbacks.

File: elements/pfe-progress-stepper/pfe-progress-step.ts (hunk 1 above)
Comment (brianferry) [reply elements/pfe-progress-stepper/pfe-progress-step.ts]: Looks good on my side now, thank you!

**Review by zeroedin — APPROVED** [review]
Lemongrass Ginger Tofu Marinade

**Pull request merged**